.travis.yml

```diff
@@ -6,11 +6,17 @@ node_js:
 
 matrix:
   fast_finish: true
+  include:
+    - node_js: 7 # Version used to deploy to npm registry
+      env: BUILD_ENV=production
 
 cache:
   directories:
     - ~/.npm
 
+before_script:
+  - NODE_ENV=$BUILD_ENV npm run build
+
```

Comment:
Correct, if `BUILD_ENV` isn't defined, it runs `NODE_ENV= npm run build` which is similar as setting an empty environment (i.e. a not production).